zerver/views/registration.py

```diff
@@ -116,14 +116,24 @@ def get_prereg_key_and_redirect(
     accidentally adding an extra character after pasting).
     """
     try:
-        check_prereg_key(request, confirmation_key)
+        prereg_user = check_prereg_key(request, confirmation_key)
     except ConfirmationKeyError as e:
         return render_confirmation_key_error(request, e)
 
+    realm_creation = prereg_user.realm_creation
+
+    registration_url = "/accounts/register/"
+    if realm_creation:
+        registration_url = "/realm/register/"
```

Comment:
It would be cleaner to use `reverse("accounts_register")`/`reverse("realm_register")` rather than hard-coding the urls I think

Reply:
Makes sense. Done!